0005-cmx2cdp-protocol/README.md

```diff
@@ -0,0 +1,207 @@
+- `timestamp` (number, required) - refer to [Kaa BB: messaging IPC](/0003-messaging-ipc/README.md) documentation for description.
+- `timeout` (number, required) - refer to [Kaa BB: messaging IPC](/0003-messaging-ipc/README.md) documentation for description.
+- `appVersionToken` (string, required) - application version to which configuration schema is applicable.
+- `endpointId` (string, required) - unique identifier of endpoint to which configuration schema is applicable.
+- `configVersion` (int, optional) - version of endpoint configuration. If not present, response message will
+hold configuration with latest version.
+
+Example:
+
+```
+{
+  "correlationId": "07d78e95-2c4d-4899-957c-b9e5a3701fbb",
+  "timestamp": 1490303342158,
+  "timeout": 3000,
+  "appVersionToken": "39774993-a426-4092-9e38-02ec213272d0",
+  "endpointId": "b197e391-1d13-403b-83f5-87bdd44888cf",
+  "configVersion": 7
+}
+```
+
+_“config-response”_ message structure:
+
+- `correlationId` (string, required) - refer to [Kaa BB: messaging IPC](/0003-messaging-ipc/README.md) documentation for description.
+- `timestamp` (number, required) - refer to [Kaa BB: messaging IPC](/0003-messaging-ipc/README.md) documentation for description.
+- `timeout` (number, required) - refer to [Kaa BB: messaging IPC](/0003-messaging-ipc/README.md) documentation for description.
+- `appVersionToken` (string, required) - application version to which configuration schema is applicable.
+- `endpointId` (string, required) - endpoint unique identifier to which configuration schema is applicable.
+- `contentType` (string, required) - content type of configuration schema, e.g.: json, protobuf.
+- `configVersion` (int, required) - version of configuration schema.
+- `statusCode` (int, required) - status code that holds meaningful information about the result of inbound message processing.
+- `content` (byte[], optional) - content with configuration.
+
+Example:
+
+```
+{
+  "correlationId": "07d78e95-2c4d-4899-957c-b9e5a3701fbb",
+  "timestamp": 1490303342158,
+  "timeout": 3000,
+  "appVersionToken": "39774993-a426-4092-9e38-02ec213272d0",
+  "endpointId": "b197e391-1d13-403b-83f5-87bdd44888cf",
+  "configVersion": 7,
+  "statusCode": 0,
+  "content": {
+    "bytes": "d2FpdXJoM2pmbmxzZGtjdjg3eTg3b3cz"
+  }
+}
+```
+
+### Broadcast messaging
+
+Event-based approach as described in [Kaa BB: messaging IPC](/0003-messaging-ipc/README.md) documentation can be used as alternative, 
+thus manipulations with configuration schemas are accepted as events.
+
+#### Subject structure
+
+CMX and CDP should listen on and send broadcast messages to the next subject:
+
+    kaa.v1.events.{originator-service-instance-id}.endpoint.config.{event-type}
+
+For subject parts explanation refer to [Kaa BB: messaging IPC](/0003-messaging-ipc/README.md).
+
+There are next types of such messages:
+
+* _"updated"_ message
```

Comment:
It's always a good idea to explain the purpose of whatever term you're introducing. From reading these type names I don't get a faint idea of what they are.
You should also provide rationale on why there are two broadcast types rather than one.